spring-boot/admin/src/main/java/eu/coatrack/admin/controllers/GatewayApiController.java

```diff
@@ -51,4 +66,26 @@ public ApiKey findApiKeyEntityByApiKeyValue(@RequestParam("keyValue") String api
     public ServiceApi findServiceByApiKeyValue(@RequestParam("apiKeyValue") String apiKeyValue) {
         return serviceApiRepository.findByApiKeyValue(apiKeyValue);
     }
+
+    @GetMapping( "/api/gateways/{gatewayId}/receiveApiKeyList")
```

Comment:
Just a reminder, to be discussed: 

each gateway should only be allowed to get the api key for "its own services", so technically the given gateway ID needs to match the gateway that is calling. Maybe we can use a filter condition on the repository call (to be checked if this would work)?

Reply:
Solved.